drivers/include/at86rf215.h

```diff
@@ -106,34 +106,39 @@ enum {
  * @name    Channel configuration
  * @{
  */
-#define AT86RF215_DEFAULT_CHANNEL        (CONFIG_IEEE802154_DEFAULT_CHANNEL)
-#define AT86RF215_DEFAULT_SUBGHZ_CHANNEL (CONFIG_IEEE802154_DEFAULT_SUBGHZ_CHANNEL)
+#ifndef CONFIG_AT86RF215_DEFAULT_CHANNEL
```

Comment:
It would actually be good to add these to the `config_drivers_netdev` doc group, and also expose them in the Kconfig that already exists for the driver.

Reply:
I'm wondering: (how) can I make it so that I can select between different values for a variable?

e.g. instead of having 

```C
#define CONFIG_AT86RF215_DEFAULT_PHY_MODE      (IEEE802154_PHY_OQPSK)
```

I would now do

```Kconfig
choice
    prompt "Default Modulation"

config AT86RF215_DEFAULT_LEGACY_OQPSK
    bool "legacy O-QPSK"
    help
        O-QPSK compatible with IEEE 802.15.4-2003 devices

config AT86RF215_DEFAULT_MR_OQPSK
    bool "MR-O-QPSK"
    help
        MR-O-QPSK according to IEEE 802.15.4g

config AT86RF215_DEFAULT_MR_OFDM
    bool "MR-OFDM"
    help
        MR-O-OFDM according to IEEE 802.15.4g

endchoice
```

But for things like 

```C
#define CONFIG_AT86RF215_DEFAULT_MR_OQPSK_CHIPS (AT86RF215_FCHIP_1000)
```

I'd like to chose between defines.
Or is that the wrong approach?

Reply:
If I understand what you mean correctly, we would keep the Kconfig model as a choice, and do the translation in the header file. An example:
https://github.com/RIOT-OS/RIOT/blob/d69ef1367b87495646685a20862f07fae249d725/sys/usb/usbus/Kconfig#L23-L43

https://github.com/RIOT-OS/RIOT/blob/d69ef1367b87495646685a20862f07fae249d725/sys/include/usb/usbus.h#L83-L96

Reply:
Yea that will work.

I tried setting MR-O-QPSK and MR-OFDM as default in `make menuconfig` with different options and they were applied correctly.

I did not add extra config options to KConfig for things that were already configured by `CONFIG_IEEE802154_` options.

In the future KConfig should probably also gain the ability to select which modes should be compiled in (currently selected by the pseudo-modules `netdev_ieee802154_mr_oqpsk`, `netdev_ieee802154_mr_ofdm` and `netdev_ieee802154_oqpsk`), but that would be for another PR.

I'd like to keep this one small so I can base MR-FSK off it rather soon.